src/Sarif.Converters/PREFastConverter.cs

```diff
@@ -0,0 +1,193 @@
+            var locations = new List<AnnotatedCodeLocation>();
+            bool pathUsesKeyEvents = defect.Path.SFAs.Any(x => !string.IsNullOrWhiteSpace(x?.KeyEvent?.Id));
+
+            foreach (var sfa in defect.Path.SFAs)
+            {
+                var reg = new Region()
+                {
+                    StartColumn = sfa.Column + 1,
+                    StartLine = sfa.Line
+                };
+
+                Uri uri = new Uri($"{sfa.FilePath}{sfa.FileName}", UriKind.Relative);
+                PhysicalLocation fileLocation = new PhysicalLocation(uri: uri, uriBaseId: null, region: reg);
+                var annotatedCodeLocation = new AnnotatedCodeLocation
+                {
+                    PhysicalLocation = fileLocation,
+                    Step = ++step
+                };
+
+                if (pathUsesKeyEvents)
+                {
+                    if (string.IsNullOrWhiteSpace(sfa.KeyEvent?.Id))
+                    {
+                        annotatedCodeLocation.Importance = AnnotatedCodeLocationImportance.Unimportant;
+                    }
+                    else
+                    {
+                        annotatedCodeLocation.SetProperty("keyEventId", sfa.KeyEvent.Id);
+                        if (Enum.TryParse(sfa.KeyEvent.Kind, true, out AnnotatedCodeLocationKind kind))
+                        {
+                            annotatedCodeLocation.Kind = kind;
+                        }
+
+                        if (Enum.TryParse(sfa.KeyEvent.Importance, true, out AnnotatedCodeLocationImportance importance))
+                        {
+                            annotatedCodeLocation.Importance = importance;
+                        }
+
+                        if (!string.IsNullOrWhiteSpace(sfa.KeyEvent.Message))
+                        {
+                            annotatedCodeLocation.Message = sfa.KeyEvent.Message;
+                        }
+                    }
+                }
+
+                locations.Add(annotatedCodeLocation);
+            }
+
+            var codeFlow = new CodeFlow()
+            {
+                Locations = locations
+            };
+
+            result.CodeFlows = new List<CodeFlow>()
+            {
+                codeFlow
+            };
+        }
+
+        private void ExtractCategories(Defect defect, Result result)
+        {
+            if (defect.Category != null)
+            {
+                foreach (var keyValuePair in defect.Category)
+                {
+                    string category = keyValuePair.Key;
+                    if (categories.ContainsKey(category))
+                    {
+                        category = categories[category];
+                    }
+
+                    result.SetProperty(category, keyValuePair.Value);
+                }
+            }
+        }
+
+        private string RemoveLegacyNewLine(string content)
+        {
+            return content.Replace("PREFAST_NEWLINE\n", "");
```

Comment:
PREfast files use this as their line break.  Remove it.

Reply:
We generally use `string.Empty` instead of `""`.

Reply:
fixed

---
In reply to: [146681613](https://github.com/Microsoft/sarif-sdk/pull/657#discussion_r146681613) [](ancestors = 146681613)